addons/account/models/account_journal_dashboard.py

```diff
@@ -420,6 +420,9 @@ def _fill_bank_cash_dashboard_data(self, dashboard_data):
             has_outstanding, outstanding_pay_account_balance = outstanding_pay_account_balances[journal.id]
             to_check_balance, number_to_check = to_check.get(journal, (0, 0))
             misc_balance, number_misc = misc_totals.get(journal.default_account_id, (0, 0))
+            if journal.currency_id:
+                company_currency = self.env['res.currency'].browse(journal.company_id.sudo().currency_id.id)
```

Comment:
Why do we have to browse a record that we already have?
Why are we using `sudo`?